app/models/ptable.rb

```diff
@@ -2,29 +2,55 @@
 # It supports both static partition maps and dynamic scripts that create partition tables on-the-fly
 # A host object may contain a reference to one of these ptables or, alternatively, it may contain a
 # modified version of one of these in textual form
-class Ptable < ActiveRecord::Base
+class Ptable < Template
   include Authorizable
   extend FriendlyId
   friendly_id :name
   include Parameterizable::ByIdName
   include ValidateOsFamily
+
   audited :allow_mass_assignment => true
+  self.auditing_enabled = !Foreman.in_rake?('db:migrate')
+  has_many :audits, :as => :auditable, :class_name => Audited.audit_class.name
 
-  validates_lengths_from_database
   before_destroy EnsureNotUsedBy.new(:hosts, :hostgroups)
   has_many_hosts
   has_many :hostgroups
   has_and_belongs_to_many :operatingsystems
   validates :layout, :presence => true
-  validates :name, :presence => true, :uniqueness => true
-  default_scope lambda { order('ptables.name') }
+  validates :name, :uniqueness => true
   validate_inclusion_in_families :os_family
 
-  scoped_search :on => :name, :complete_value => true, :default_order => true
-  scoped_search :on => :layout, :complete_value => false
-  scoped_search :on => :os_family, :rename => "family", :complete_value => :true
+  # these can't be shared in parent class, scoped search can't handle STI properly
+  # tested with scoped_search 3.2.0
+  include Taxonomix
```

Comment:
Operatingsystem is not Taxable, but maybe we could take all taxonomies from hosts for a given partition table. This way we'd keep minimal assignment to keep it working. On the other hand, before this patch user could select any partition table so I think all taxonomies could be better option.